From 315d1deefb519c54b53d9716d303505051daebcf Mon Sep 17 00:00:00 2001
From: Alfred Neumayer <dev.beidl@gmail.com>
Date: Sun, 12 Dec 2021 17:24:57 +0100
Subject: [PATCH] (halium) init.rc: Create directory for trust-store socket

This enables the minimediaservice process to create the socket
in a well-known location for consumption by trust-store.

Change-Id: Iac114c1b0d92afb54d39ad314c6b521d6c088dc1
---
 rootdir/init.rc | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/rootdir/init.rc b/rootdir/init.rc
index c2426ce..da57d5f 100644
--- a/rootdir/init.rc
+++ b/rootdir/init.rc
@@ -400,6 +400,11 @@ on init
     start hwservicemanager
     start vndservicemanager
 
+    # Create directory necessary for the CameraService trust-store socket
+    mkdir /dev/socket/camera_service
+    chown root camera /dev/socket/camera_service
+    chmod 0775 /dev/socket/camera_service
+
 # Healthd can trigger a full boot from charger mode by signaling this
 # property when the power button is held.
 on property:sys.boot_from_charger_mode=1
-- 
2.32.1 (Apple Git-133)

